// Copyright 2024 Flower Labs GmbH. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
// ==============================================================================

syntax = "proto3";

package flwr.proto;

import "flwr/proto/node.proto";

message Fab {
  // This field is the hash of the data field. It is used to identify the data.
  // The hash is calculated using the SHA-256 algorithm and is represented as a
  // hex string (sha256hex).
  string hash_str = 1;
  // This field contains the fab file contents a one bytes blob.
  bytes content = 2;
  // Metadata.
  map<string, string> meta = 3;
}

message GetFabRequest {
  Node node = 1;
  string hash_str = 2;
  uint64 run_id = 3;
}
message GetFabResponse { Fab fab = 1; }
